### PR TITLE
fix: fix inert mechanism for overlays

### LINKED ISSUE #2969: [Bug]: data-sbb-inert blocks interaction for dialogs
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/sbb-design-systems/lyne-components/blob/main/docs/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/sbb-design-systems/lyne-components/blob/main/docs/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/orgs/sbb-design-systems/projects/4) for an issue that matches the one I want to file, without success.

### Bug type

Functionality

### It affects the following packages

Components

### Which version of Lyne Components are you using

1.2.0

### Operating system

Windows 10

### Browser / Browser version

Chrome

### Input mode

None

### Additional settings

_No response_

### What happened?

the "data-sbb-inert" or "inert" attribute is blocking the interaction for new opened dialogs.

### Steps to reproduce the issue

1.  use a sbb-dialog is hierarchically at the top of the DOM with an action button.
2. The action button opens a new sbb-dialog, which is positioned hierarchically at the bottom of the DOM.
3. The newly opened dialog is displayed, but its interaction is blocked by the data-sbb-inert attribute of the first dialog.

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jeripeierSBB**
Closes #2969

### CODE REVIEW

File: src/elements/core/overlay/inert.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
+const IGNORED_ELEMENTS = ['script', 'head', 'template'];
```
Comment (WalkingOS) [inline src/elements/core/overlay/inert.ts:1]: is there maybe more elements which should be ignored? Maybe <link/> <style/> <meta/> ? Are we assuming here that those elements are usually inside the <head/> element?

Comment (jeripeierSBB) [reply src/elements/core/overlay/inert.ts:1]: Thanks for the hint, we can add style as well. These elements in the list have no direct impact on the inert mechanism as they are more meta-like. Conversely, there wouldn't be any change if they received inert, so at the end it doesn't change much.

**Review by dauriamarco — APPROVED** [review]
Thank you very much! LGTM :)

File: src/elements/core/controllers/inert-controller.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  }
+
+  public hostConnected(): void {
+    if (this._host.isOpen) {
+      this.activate();
+    }
+  }
+
+  public hostDisconnected(): void {
+    this.deactivate(true);
```
Comment (kyubisation) [inline src/elements/core/controllers/inert-controller.ts:25]: As discussed

**Review by kyubisation — APPROVED** [review]
LGTM 👍

**Pull request merged**